### ISSUE TITLE
VScode is connecting to wrong remote ssh server

### ISSUE THREAD

**Issue by deepio-oc** [microsoft/vscode-remote-release#9626]
- VS Code Version:  1.85.1
- OS Version: Mac Sonoma 14.3

Steps to Reproduce:

1. `code --folder-uri vscode-remote://ssh-remote+<username>@<IP1>:<PORT1>/home/<username>`

2. `code --folder-uri vscode-remote://ssh-remote+<username>@<IP1>:<PORT2>/home/<username>`

This happens in kubernetes environment where each VSCode is running in a different pod but the node IP(IP1) is same for both the pods but node ports (PORT1 & PORT2) are different.

Both commands end up connecting to same vscode pod (even though port number is different). SSH works fine and correctly connects to different pods.

**Comment by roblourens** [comment]
I don't think we actually parse ports using this syntax. I recommend adding this host to your ssh config and connecting just using the name from the ssh config.
